scripts/db_migrator.py

```diff
@@ -96,6 +100,89 @@ def migrate_interface_table(self):
+                    buffer_pool_conf = config_db['BUFFER_POOL']
+            except IOError:
+                log_error("failed to open config_db.json file, skip migration")
+                return False
+        else:
+            # Try to get related info from DB(in warm reboot case DB will be retored)
+            buffer_pool_conf = self.configDB.get_table('BUFFER_POOL')
+            device_data = self.configDB.get_table('DEVICE_METADATA')
+            if 'localhost' in device_data.keys():
+                hwsku = device_data['localhost']['hwsku']
+                platform = device_data['localhost']['platform']
+            else:
+                log_error("Trying to get data from DB but it's empty, skip migration")
+                return False
+
+        # Get current buffer pool configuration, only migrate configration which 
+        # with default values, if it's not default, leave it as is.
+        pool_size_in_db_list = []
+        pools_in_db = buffer_pool_conf.keys()
+
+        # Buffer pool numbers is different with default, don't need migrate
+        if len(pools_in_db) != len(buffer_pools):
+            return True
+
+        # If some buffer pool is not default ones, don't need migrate
+        for buffer_pool in buffer_pools:
+            if buffer_pool not in pools_in_db:
+                return True
+            pool_size_in_db_list.append({buffer_pool: buffer_pool_conf[buffer_pool]['size']})
+        
+        # To check if the buffer pool size is equal to old default values
+        if pool_size_in_db_list == spc1_t0_default_value or pool_size_in_db_list == spc1_t1_default_value \
+            or pool_size_in_db_list == spc2_t0_default_value or pool_size_in_db_list == spc2_t1_default_value:
+            # Generate buffer configuration from latest template
+            tmp_json_file = "/tmp/mlnx_buffers_all.json"
+            device_buffer_template_path = os.path.join("/usr/share/sonic/device/", platform, hwsku, 'buffers.json.j2')
+            command = "sonic-cfggen -d -t {} >{}".format(device_buffer_template_path, tmp_json_file)
+            try:
+                subprocess.check_call(command, shell=True)
+            except:
+                log_error("failed to generate new mlnx buffer configuration, skip migration")
+                return False
+
+            # Extract the new buffer pool size configuration from above tmp file and update DB
+            try:
+                with open(tmp_json_file) as tmp_buffer_file:
+                    buffer_json = json.load(tmp_buffer_file)
+                    for pool in buffer_pools:
+                        self.configDB.set_entry('BUFFER_POOL', pool, buffer_json['BUFFER_POOL'][pool])
+                    log_info("Successfully migrate mlnx buffer pool size to the latest.")
+                    return True
+            except IOError:
+                log_error("failed to open tmp mlnx buffer json configuration file, skip migration")
+                return False
```

Comment:
Same comment here. Can you make it not fail?

Reply:
as above comments